dima/react-movies/src/pages/mainpage/mainpage.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from "react";
+import ContentContainer from "../../components/contentContainer";
+import Header from "../../components/header";
+import SearchBar from "../../components/searchBar";
+import MoviesContainer from "../../components/moviesContainer";
+import ErrorBoundary from "../../components/errorBoundary";
+import MoviesResult from "../../components/moviesResult";
+import SortFilter from "../../components/sortFilter";
+import Footer from "../../components/footer";
+import QueryString from "query-string";
+import { useHistory, useLocation } from "react-router-dom";
+import SortProperty from "../../enums/SortProperty";
+import ParamsToPush from "../../interfaces/paramsToPush";
+
+import "./mainpage.css";
+import { useDispatch, useSelector } from "react-redux";
+import { resetMovies, fetchMovies } from "../../redux/moviesReducer";
+import { StateInterface } from "../../interfaces/stateInterface";
+import MoviesDataInterface from "../../interfaces/moviesDataInterface";
+
+const MainPage = () => {
+  const [currentSortType, setCurrentSortType] = useState(SortProperty.date);
+  let history = useHistory();
+  let location = useLocation();
+  const dispatch = useDispatch();
+  const movies = useSelector<StateInterface, MoviesDataInterface>(
+    (state) => state.movies
+  );
+
+  const loading = useSelector<StateInterface, boolean>(
+    (state) => state.loadingMovies
+  );
+
+  useEffect(() => {
+    const params = QueryString.parse(location.search);
+    const compareSortFromUrlToState = (
+      sortBy: string | string[] | null
+    ): boolean => sortBy !== currentSortType;
+
+    const fetchMoviesLocal = (pageNum: number): void => {
+      let defaultParams = {
+        limit: 9,
+        sortBy: "release_date",
+        sortOrder: "desc",
+        offset: 0,
+      };
+      const oldParamsObj = QueryString.parse(location.search);
+      const { sortBy } = oldParamsObj;
+      if (sortBy && compareSortFromUrlToState(sortBy))
+        setCurrentSortType(
+          sortBy === SortProperty.date ? SortProperty.date : SortProperty.rating
+        );
```

Comment:
better to always wrap your code in brackets, even when there is only one operation inside.